src/Container.js

```diff
@@ -78,6 +91,18 @@ export default class Container {
         };
         this._instanceCache[name] = instance;
     }
+    registerSingleton(name, instance, isExternallyOwned = true) {
```

Comment:
Same here, why would you expect isExternallyOwned to be true by default?  In fact, why make it optional?